### ISSUE TITLE
Validator: Validator fails for HISTOLOGY attribute

### ISSUE THREAD

**Issue by ritikakundra** [cBioPortal/datahub#51]
<img width="1159" alt="screen shot 2017-06-14 at 12 27 43 pm" src="https://user-images.githubusercontent.com/12007862/27143664-2642ad08-50fd-11e7-9330-33a5378f8db1.png">

@pieterlukasse @zheins 

It should not fail for an attribute. Also in our config file, it is a patient based attribute.

**Comment by ritikakundra** [comment]
Not sure who to assign

**Comment by pieterlukasse** [comment]
solution discussed today: @ritikakundra will list the attributes that are patient specific and thus need to be moved. 
@sandertan will adjust the validator and log issues for the frontend team to fix if any of these attributes is hard coded in the frontend.

**Comment by ritikakundra** [comment]
These are patient based attributes:
SERUM_PSA
TUMOR_GRADE
HISTOLOGY
TUMOR_SITE
KNOWN_MOLECULAR_CLASSIFIER


And:
TYPE_OF_CANCER: Overrides CANCER_TYPE in the header
DETAILED_CANCER_TYPE: Overrides CANCER_TYPE_DETAILED in the header
are not in out configuration sheet of clinical attributes. Any idea why we have these?

**Comment by sandertan** [comment]
Before we change the validator, we should make sure this will not break anything in the frontend. 

Together with @fedde-s we identified that HISTOLOGY and KNOWN_MOLECULAR_CLASSIFIER are used in the patient view for a specifying the order of attributes: https://github.com/cBioPortal/cbioportal-frontend/blob/a1064b8bcebef20268a2baa667a69cac03bda714/src/pages/patientView/patientHeader/style/clinicalAttributes.scss

TUMOR_SITE also seems to be a patient attribute, according to https://github.com/cBioPortal/cbioportal-frontend/blob/a1064b8bcebef20268a2baa667a69cac03bda714/src/pages/patientView/clinicalInformation/lib/clinicalAttributesUtil.js

We should also update (and maybe recheck) the list of documented attributes on https://github.com/cBioPortal/cbioportal/blob/v1.7.1/docs/File-Formats.md#the-samples-file

**Comment by fedde-s** [comment]
Actually, `clinicalAttributesUtil.js` seems to suggest that `TUMOR_SITE` describes something about a *sample*, where for metastatic samples it would be overridden by `METASTATIC_SITE`, and for primary tumour samples it would be overridden by `PRIMARY_SITE`. Whether a sample is primary, metastatic or something else is defined by a specific list of recognised keywords in either `SAMPLE_CLASS`, `SAMPLE_TYPE`, `TUMOR_TISSUE_SITE` or `TUMOR_TYPE`, all treated as synonyms.

**Comment by fedde-s** [comment]
The clinical information derived as above is used in https://github.com/cBioPortal/cbioportal-frontend/blob/a1064b8bcebef20268a2baa667a69cac03bda714/src/pages/patientView/sampleManager.tsx, a module which is imported in various other .tsx files.

**Comment by sandertan** [comment]
This old commit message suggests TUMOR_TYPE is a sample attribute: 

https://github.com/cBioPortal/cbioportal/commit/88892a126edc9f89c7dfd68a9ac55549709b817d

**Comment by pieterlukasse** [comment]
@inodb can you help us with the frontend part? Maybe some hard coded items can be removed?

**Comment by inodb** [comment]
@sandertan @pieterlukasse Changing these attributes should be fine. They should just show up after the patient id instead of the sample id. I think all these attributes are still in our data, so would rather not remove these hard coded items.

As a side note re documentation. You could perhaps point directly from the documentation to the source code file (I think most people understand css/sass). To make sure the docs stay up to date you could add a simple bash script that's run on travis that makes sure the file still exists. Maybe overkill, but just putting it out there.

**Comment by fedde-s** [comment]
@inodb, it does seem like `TUMOR_SITE` has a special meaning to the JavaScript code that I linked, that may not be as useful on the patient level.

**Comment by inodb** [comment]
@fedde-s Good point. Seems like you might want to have `TUMOR_SITE` both at sample and patient level. I'll defer this to @ritikakundra. What are your thoughts?

**Comment by fedde-s** [comment]
The database schema and legacy data file formats do not allow having two attributes with the same ID in one study.

**Comment by inodb** [comment]
@fedde-s ah I see. Yeah there seem to be some studies in our internal database that have `TUMOR_SITE` as sample attribute, majority have it as patient. I guess we could get rid of the `TUMOR_SITE` check in the code if all has been moved over. Seems like `PRIMARY_SITE` and `METASTATIC_SITE` are indeed mostly used as sample attributes, so that should be sufficient.

**Comment by fedde-s** [comment]
I always imagined that it was the exact other way around based on the code… Which attribute indicates the site of a sample that is neither the primary tumour nor a metastasis, if not `TUMOR_SITE`?

**Comment by inodb** [comment]
@ritikakundra Do you know if there is a clinical attribute to indicate a site of a sample w/o using `PRIMARY_SITE`, `METASTATIC_SITE` or `TUMOR_SITE`?

**Comment by pieterlukasse** [comment]
I am happy this is being fixed :+1:  I added a PR to help prevent such issues in the future: https://github.com/cBioPortal/cbioportal-frontend/pull/987 . It is not a hard check, but should help make frontend developers better aware of the importance of updating validation and documentation when implementing a dependency on a clinical attribute.

**Comment by CuriusScientist** [comment]
@sandertan  I agree with you I am facing the same problem and don't know most of the terminologies used by cBioPortal. In some cases, they are taken from other tools, edited and abbreviated so there is no way to find what came from where.

**Comment by fedde-s** [comment]
What's the reason for closing this? Has it been fixed (as defined in this thread) by #3910?

**Comment by fedde-s** [comment]
Ah, I see, that's a yes. @pieterlukasse said this:
> @sandertan will adjust the validator and log issues for the frontend team to fix if any of these attributes is hard coded in the frontend.
